File: problems/1103-distribute-candies-to-people/solution.py
class Solution:
    def distributeCandies(self, candies: int, num_people: int) -> List[int]:
        ans = [0] * num_people
        i = 0
        candy_to_give = 1
        while candies > 0:
            ans[i % num_people] += min(candies, candy_to_give)
            candies -= candy_to_give
            candy_to_give += 1
            i += 1
        return ans

File: problems/1103-distribute-candies-to-people/solution.java
class Solution {
    public int[] distributeCandies(int candies, int num_people) {
        int[] ans = new int[num_people];
        int i = 0;
        int give = 1;
        while (candies > 0) {
            ans[i % num_people] += Math.min(candies, give);
            candies -= Math.min(candies, give);
            give++;
            i++;
        }
        return ans;
    }
}